src/ess/amor/load.py

```diff
@@ -30,98 +32,56 @@ def load_detector(
 
 
 def load_events(
-    detector: LoadedNeXusDetector[RunType], detector_rotation: DetectorRotation[RunType]
+    detector: LoadedNeXusDetector[RunType],
+    detector_rotation: DetectorRotation[RunType],
+    sample_rotation: SampleRotation[RunType],
+    chopper_phase: ChopperPhase[RunType],
+    chopper_frequency: ChopperFrequency[RunType],
+    chopper_distance: ChopperDistance[RunType],
+    chopper_separation: ChopperSeparation[RunType],
+    sample_size: SampleSize[RunType],
+    beam_size: BeamSize[RunType],
+    angle_to_center_of_beam: AngleCenterOfIncomingToHorizon[RunType],
 ) -> RawDetectorData[RunType]:
-    detector_numbers = sc.arange(
-        "event_id",
-        start=1,
-        stop=(Detector.nBlades * Detector.nWires * Detector.nStripes).value + 1,
-        unit=None,
-        dtype="int32",
-    )
+    detector_numbers = pixel_coordinates_in_detector_system()
     data = (
-        detector['data']
+        detector["data"]
         .bins.constituents["data"]
-        .group(detector_numbers)
-        .fold(
-            "event_id",
-            sizes={
-                "blade": Detector.nBlades,
-                "wire": Detector.nWires,
-                "stripe": Detector.nStripes,
-            },
-        )
+        .group(detector_numbers.data.flatten(to='event_id'))
+        .fold("event_id", sizes=detector_numbers.sizes)
     )
-    # Recent versions of scippnexus no longer add variances for events by default, so
-    # we add them here if they are missing.
+    data.coords.update(detector_numbers.coords)
+
     if data.bins.constituents["data"].data.variances is None:
         data.bins.constituents["data"].data.variances = data.bins.constituents[
             "data"
         ].data.values
 
-    pixel_inds = sc.array(dims=data.dims, values=data.coords["event_id"].values - 1)
-    position, angle_from_center_of_beam = pixel_coordinate_in_lab_frame(
-        pixelID=pixel_inds, nu=detector_rotation
-    )
-    data.coords["position"] = position.to(unit="m", copy=False)
-    data.coords["angle_from_center_of_beam"] = angle_from_center_of_beam
+    data.coords["sample_rotation"] = sample_rotation.to(unit='rad')
+    data.coords["detector_rotation"] = detector_rotation.to(unit='rad')
+    data.coords["chopper_phase"] = chopper_phase
+    data.coords["chopper_frequency"] = chopper_frequency
+    data.coords["chopper_separation"] = sc.abs(chopper_separation)
+    data.coords["L1"] = sc.abs(chopper_distance)
+    data.coords["L2"] = data.coords['distance_in_detector'] + Detector.distance
```

Comment:
As a reminder, this should be done by adapting the `bemaline` graph for coord transforms from ScippNeutron.

Reply:
The `beamline` graph defines the following quantities:

* incident_beam is the vector of incoming neutrons on the sample
* L1
* scattered_beam is the vector of neutrons that were scattered off the sample 
* L2 
* Ltotal

But some of those are not used in this workflow, and some are to be replaced anyway:

* incident_beam - this quantity is not needed in the workflow and the definition in scippneutron is not correct in focused reflectometry.
* L1 - should be replaced 
* scattered_beam is - this quantity is not used in the workflow
* L2 - should be replaced
* Ltotal - this is used in the workflow

In the end, the only useful quantity we get from the graph is `Ltotal`. Hardly something that's difficult to compute or likely to change!

For this reason I don't think it's necessary to use that graph here. Adapting it to fit here is likely more code than saved by using it.

I think it's good to take into consideration that it becomes harder for a user to follow what transformations were applied if they have to go and look in a different package to find the definitions. And while that's sometimes unavoidable, it doesn't seem like it is here.

Reply:
Take a look at the LoKI workflow: https://scipp.github.io/esssans/user-guide/loki/loki-iofq.html It has a provider for its coord transform graphs. This way, we can have the same code do the actual transforms for multiple instruments. We only need to replace the graphs.